### PR TITLE
doc: update outdated architectural doc

### PR DESCRIPTION
**Pull request by VGalaxies**
> inspired by this https://github.com/apache/incubator-hugegraph-doc/pull/268#discussion_r1230459687 and the [task](https://github.com/orgs/apache/projects/263/views/1?pane=issue&itemId=29985963)

I found that the components in the current architecture document, such as SparkGraphX and HugeGraph-Studio, are outdated. Therefore, I have redrawn an architecture diagram and updated the corresponding documentation.

Currently, I have only drawn an overall architecture diagram, and further details can be refined, such as memory data structures, disk storage structures, etc., which need to be updated later.

> EN doc will be synchronized later.

### CODE REVIEW

File: content/cn/docs/guides/architectural.md [hunk 1]
```diff
@@ -6,25 +6,22 @@ weight: 1
 
 ### 1 概述
 
-作为一款通用的图数据库产品，HugeGraph需具备图数据的基本功能，如下图所示。HugeGraph包括三个层次的功能，分别是存储层、计算层和用户接口层。 HugeGraph支持OLTP和OLAP两种图计算类型，其中OLTP实现了[Apache TinkerPop3](https://tinkerpop.apache.org)框架，并支持[Gremlin](https://tinkerpop.apache.org/gremlin.html)查询语言。 OLAP计算是基于SparkGraphX实现。
+作为一款通用的图数据库产品，HugeGraph 需具备图数据的基本功能，如下图所示。
 
 <div style="text-align: center;">
-  <img src="/docs/images/design/architectural-overview.png" alt="image">
+  <img src="/docs/images/design/architectural-revised.png" alt="image">
```
Comment (imbajin) [inline content/cn/docs/guides/architectural.md:12]: use this as the temp arch pic

File: content/cn/docs/guides/architectural.md [hunk 2]
```diff
@@ -6,25 +6,22 @@ weight: 1
 
 ### 1 概述
 
-作为一款通用的图数据库产品，HugeGraph需具备图数据的基本功能，如下图所示。HugeGraph包括三个层次的功能，分别是存储层、计算层和用户接口层。 HugeGraph支持OLTP和OLAP两种图计算类型，其中OLTP实现了[Apache TinkerPop3](https://tinkerpop.apache.org)框架，并支持[Gremlin](https://tinkerpop.apache.org/gremlin.html)查询语言。 OLAP计算是基于SparkGraphX实现。
+作为一款通用的图数据库产品，HugeGraph 需具备图数据的基本功能，如下图所示。
```
Comment (imbajin) [inline content/cn/docs/guides/architectural.md:9]: 这里可以参考 server 的 readme 加一点说明，比如我们现在也支持cypher，简单概括即可，另外这部分内容是不是与 doc 的 introduction 页面有一部分重了，也可以考虑引用它直接

**Review by imbajin — COMMENTED** [review]
fine to sync en version

**Comment by VGalaxies** [comment]
@imbajin I noticed that each image in the documentation is stored separately in the cn and en folders, but it seems that the images are not being handled for i18n currently. Is this necessary?

File: content/en/docs/quickstart/hugegraph-server.md [hunk 3]
```diff
@@ -20,7 +20,7 @@ Consider use Java 11 to run `HugeGraph-Server` (also compatible with Java 8 now)
 
 **Be sure to execute the `java -version` command to check the jdk version before reading**
 
-#### 2.2 Install GCC-4.3(GLIBCXX_3.4.10) or higher version (optional)
+#### 2.2 Install GCC-4.3 (GLIBCXX_3.4.10) or higher version (optional)
 
 If you are using the `RocksDB` backend, be sure to execute the `gcc --version` command to check its version; Not required if you are using other backends.
```
Comment (imbajin) [inline content/en/docs/quickstart/hugegraph-server.md:25]: remove or simplify this section（gcc4.3 is too old to say，maybe it's enough to leave one line to remind user?）so as the CN

Comment (VGalaxies) [reply content/en/docs/quickstart/hugegraph-server.md]: Removed. The hint has already been given in the `Startup - RocksDB` section.

**Comment by imbajin** [comment]
> @imbajin I noticed that each image in the documentation is stored separately in the cn and en folders, but it seems that the images are not being handled for i18n currently. Is this necessary?

yep, but no one trans them now. we could enhance them in a separate PR.

and it's fine to sync EN doc

**Review by imbajin — APPROVED** [review]
LGTM

**Pull request merged**